test/python/transpiler/test_permrowcol.py

```diff
@@ -521,8 +522,22 @@ def test_perm_row_col_does_correct_permutation_matrix(self):
         )
 
         instance = permrowcol.perm_row_col(parity_mat)
+        mat = np.array(LinearFunction(instance[0]).params[0]).astype(int)
         self.assertEqual(np.array_equal(parity_mat, correct_permutation_matrix), True)
 
+    def test_common_case_with_complete_graph(self):
+        """Test common_case with complete graph"""
+        n = 6
+        parity_mat = build_random_parity_matrix(42, n, 60)
+        coupling_list = [(i, j) for i in range(n) for j in range(n) if i != j]
+        original_parity_map = parity_mat.copy()
+        coupling = CouplingMap(coupling_list)
+        permrowcol = PermRowCol(coupling)
+        circuit, perm = permrowcol.perm_row_col(parity_mat)
+        circuit_matrix = LinearFunction(circuit).linear.astype(int).T
```

Comment:
You don't want to use the transpose of the parity matrix, but its inverse. Here you can easily do that by reversing the order of operations of the output circuit:
`LinearFunction(circuit.reverse()).linear`
Then, the two matrices should cancel each other out.

Given that you are merging this code, I am assuming this test passes eventhough it shouldn't.
This means that the permrowcol assumes the transposed representation with respect to the Qiskit Library. Fixing that is a problem for a later sprint, but it should be taken into account in the run method of PermRowColSynthesis.